### DISCUSSION TITLE
batching notifications to avoid rerenders

### DISCUSSION THREAD

**Discussion by jindal-git** [TanStack/query#983] [General]
We use react-query to manage server side state, so we have multiple hooks in the hooks tree fetching the same query. When the query resolves, we get one render for every useQuery call in the hooks tree. This results in explosions of re-renders for a large hooks tree. See the minimal example below that shows 3 renders instead of 2.

https://codesandbox.io/s/hardcore-forest-5rg07?file=/src/index.js

Is this fundamental to how react-query works? Can it be improved batching notifications?

**Comment by Purus** [comment]
I am also facing this same issue.. Please let me know of any solution .

**Reply by jindal-git** [reply]
Solution is https://github.com/tannerlinsley/react-query/pull/989. 

Niek Bosch, Thank you for your lightning response! Really looking forward to it landing.
